plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineRetryStrategy.java

```diff
@@ -93,4 +97,39 @@ public boolean matchRetryCondition(SdkResponse response, Exception ex) {
         return statusCode == ERROR_CODE_TOO_MANY_REQUESTS
                 || statusCode == ERROR_CODE_INTERNAL_SERVER_ERROR;
     }
+
+    /**
+     * Handle auth exception for spring retry.
+     *
+     * @param ex Exception
+     */
+    public void handleAuthExceptionForSpringRetry(Exception ex) {
+        int retryCount = Objects.isNull(RetrySynchronizationManager.getContext())
+                ? 0 : RetrySynchronizationManager.getContext().getRetryCount();
+        log.error(ex.getMessage() + System.lineSeparator() + "Retry count:" + retryCount);
+        if (ex instanceof ClientAuthenticationFailedException) {
```

Comment:
do we really need this check? this exception is only thrown by the code below it? 

if this is not needed, we can remove from other plugins also

Reply:
The method getPrice called the method getProjectId which will throw ClientAuthenticationFailedException , it is needed
![image](https://github.com/user-attachments/assets/04ba27f6-43e9-4327-b94c-a60655ba8c55)